ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp

```diff
@@ -571,27 +571,7 @@ int window_get_cursor()
 
 bool keyboard_check_direct(int key)
 {
-   BYTE keyState[256];
-
-   if ( GetKeyboardState( keyState ) )  {
-	  for (int x = 0; x < 256; x++)
-		keyState[x] = (char) (GetKeyState(x) >> 8);
-   } else {
-      //TODO: print error message.
-	  return 0;
-   }
-
-  if (key == vk_anykey) {
-    for(int i = 0; i < 256; i++)
-      if (keyState[i] == 1) return 1;
-    return 0;
-  }
-  if (key == vk_nokey) {
-    for(int i = 0; i < 256; i++)
-      if (keyState[i] == 1) return 0;
-    return 1;
-  }
-  return keyState[key & 0xFF];
+  return GetAsyncKeyState(key) < 0;
```

Comment:
I'm torn between `< 0` and `& 0x8000`, but yeah. Good call.

Reply:
Yeah, but I don't like magic numbers. 😄